torch_geometric/loader/dynamic_batch_sampler.py

```diff
@@ -0,0 +1,108 @@
+from typing import Iterator, List, Optional
+
+import torch
+
+from torch_geometric.data import Dataset
+
+
+class DynamicBatchSampler(torch.utils.data.sampler.Sampler[List[int]]):
+    r"""Dynamically adds samples to mini-batch up to a maximum size (number of
+    nodes). When data samples have a wide range in sizes, specifying a
+    mini-batch size in terms of number of samples is not ideal and can cause
+    CUDA OOM errors.
+
+    Therefore, the number of steps per epoch is ambiguous, depending on the
+    order of the samples. By default the ``__len__()`` will be None, fine for
+    most cases but progress bars will be `infinite`. Alternatively,
+    ``num_steps`` can be supplied to cap the number of mini-batches produced
+    by the sampler.
+
+    **Usage:**
+
+    .. code-block:: python
+
+        from torch_geometric.loader import DataLoader, DynamicBatchSampler
+
+        batch_sampler = DynamicBatchSampler(10000, dataset)
```

Comment:
```suggestion
        batch_sampler = DynamicBatchSampler(dataset, num_nodes=10000)
```
Can we first pass in the data, and also rename `max_num` to `num_nodes`?

Reply:
Yes, the data argument first makes sense. For the other, I had it as `max_num` instead of `num_nodes` since you can also specify a maximum number of edges in the batch instead of nodes with the `entity` argument. Is there a more generic name? Or do you think `num_nodes` is fine?